skimage/transform/_warps.py

```diff
@@ -841,17 +841,19 @@ def warp(image, inverse_map, map_args={}, output_shape=None, order=1,
             matrix = np.linalg.inv(inverse_map.__self__.params)
 
         if matrix is not None:
-            matrix = matrix.astype(np.double)
+            matrix = matrix.astype(image.dtype)
+            ctype = 'float' if image.dtype.char == 'f' else 'double'
             if image.ndim == 2:
-                warped = _warp_fast(image, matrix,
-                                    output_shape=output_shape,
-                                    order=order, mode=mode, cval=cval)
+                warped = _warp_fast[ctype](image, matrix,
```

Comment:
this won't select things automatially? Maybe that was my mistake....

Reply:
Nop, it wasn't working without this specialization...

Reply:
🎉 very nice, thanks @rfezzani for this discovery!